cime_config/config_archive.xml

```diff
@@ -13,10 +13,18 @@
     <!-- match filenames of the form
          ic.(date string)[._optional instance number].nc[.optional tile number] -->
     <hist_file_extension>ic.[-\d+]+(\._\d*)?\.nc(\.\d*)?$</hist_file_extension>
+    <hist_file_ext_regex>h.bgc.z(\._\d*)?</hist_file_ext_regex>
+    <hist_file_ext_regex>h.bgc.native(\._\d*)?</hist_file_ext_regex>
     <hist_file_ext_regex>\w+\.\w+(\._\d*)?</hist_file_ext_regex>
     <rpointer>
       <rpointer_file>rpointer.ocn$NINST_STRING.$DATENAME</rpointer_file>
       <rpointer_content>$CASE.mom6$NINST_STRING.r.$DATENAME.nc</rpointer_content>
     </rpointer>
+      <test_file_names>
+        <tfile disposition="copy">rpointer.ocn.1976-01-01-00000</tfile>
+        <tfile disposition="move">casename.mom6.h.1976-01-01-00000.nc</tfile>
```

Comment:
MOM doesn't generate a `mom6.h` stream, and I suspect a file of this name would not be handled by archiver correctly. The non-BGC daily streams MOM6 generates in a CESM test

```
mom6.h.native.0001-01-05.nc
mom6.h.rho2.0001-01-05.nc
mom6.h.sfc.0001-01-05.nc
mom6.h.z.0001-01-05.nc
```

and there are also a few files without date stamps:

```
mom6.h.ocean_geometry.nc
mom6.h.static.nc
```

Lastly, it looks like we also compare `mom6.ic.0001-01-01.nc` though it's unclear how that file gets added to the baseline directory since it won't match `hist_file_ext_regex`.